packages/rules/src/mustSatisfyPeerDependencies.ts

```diff
@@ -523,10 +523,17 @@ export function findIntersection(a: ValidRange, b: ValidRange): ValidRange | und
   if (aIsGreaterOrEqualVersionRange) {
     const aSemVer = coerce(a)!;
     // keep every version where `bSemVer` is >= `aSemVer`
-    const compatibleBVersions = bVersions.filter((bVersion) => {
-      const bSemVer = coerce(bVersion)!;
-      return bSemVer.compare(aSemVer) !== -1;
-    });
+    // where bVersion has an operator, we need to adjust the version to take into
+    // account aVersion, e.g. if a is >=15.2.3 and b is ^15, then the intersection is actually ^15.2.3
+    const compatibleBVersions = bVersions
+      .map((bVersion) => {
+        const bSemVer = coerce(bVersion)!;
+        if (bVersion.startsWith("^") && bSemVer.major >= aSemVer.major) {
```

Comment:
can this just be `if (bSemVer.major >= aSemVer.major) { return bSemVer }` (not sure why `return bSemVer.compare(aSemVer) !== -1;` wasn't equivalent here)?